dist/expected.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import "github.com/gonum/quad"
+
+// ExpectedFixed numerically computes the expected value of the input function
+// under the probability distribution specified by the Quantiler. That
+// is, ExpectedFixed computes
+//
+//  ev ≈ E[f] = int_-inf^inf f(x) p(x) dx
+//
+// using the number of quadrature locations specified by evals. Concurrent
+// sets the number of allowed concurrent evaluations of f.
+//
+// ExpectedFixed is a wrapper around quad.Fixed. See the documentation for more
+// information.
+func ExpectedFixed(f func(float64) float64, q Quantiler, evals, concurrent int) (ev float64) {
```

Comment:
The concurrent version uses subintegrals though?  I feel like I am arguing semantics a bit, but it would result in slightly different output due to rounding, so to me that is a different algorithm.